client/my-sites/hosting/sftp-card/index.js

```diff
@@ -136,28 +178,38 @@ const SFTPCard = ( { translate, username, password, siteId, loading, disabled }
 					</tbody>
 				</table>
 			) }
-			{ loading && <Spinner /> }
+			{ isLoading && <Spinner /> }
 		</Card>
 	);
 };
 
-export default connect( ( state, { disabled } ) => {
-	const siteId = getSelectedSiteId( state );
-	let username = null;
-	let password = null;
-	let loading = null;
-
-	if ( ! disabled ) {
-		const sftpDetails = requestAtomicSFTPDetails( siteId );
-		username = get( sftpDetails, 'data.username' );
-		password = get( sftpDetails, 'data.password' );
-		loading = sftpDetails.state === 'pending';
-	}
+export default connect(
+	( state, { disabled } ) => {
+		const siteId = getSelectedSiteId( state );
+		const currentUserId = getCurrentUserId( state );
+		let username = null;
+		let password = null;
+		let loaded = null;
 
-	return {
-		siteId,
-		username,
-		password,
-		loading,
-	};
-} )( localize( SFTPCard ) );
+		if ( ! disabled ) {
+			const sftpDetails = getAtomicHostingSftpUser( state, siteId, currentUserId );
+			username = get( sftpDetails, 'username' );
+			password = get( sftpDetails, 'password' );
+			loaded = sftpDetails !== null;
+		}
+
+		return {
+			siteId,
+			currentUserId,
+			username,
+			password,
+			loaded,
+		};
+	},
+	{
+		requestSftpUser: requestAtomicSftpUser,
+		createSftpUser: createAtomicSftpUser,
+		resetSFTPPassword: resetAtomicSFTPPassword,
+		removePasswordFromState: updateAtomicSftpUser,
```

Comment:
This could be more maintainable if `removePasswordFromState` is a wrapper around `updateAtomicSftpUser` that removes the password from the state:
```suggestion
		removePasswordFromState: ( siteId, userId, username ) => updateAtomicSftpUser( siteId, userId, { username } ),
```

Otherwise, the component could use `removePasswordFromState` for actually setting a password.

Reply:
Yeh, I felt slightly uncomfortable about that direct mapping but couldn't think of another way, but the wrapper does it - thanks.